DatadogSessionReplay/Sources/Feature/RequestBuilders/Multipart/MultipartFormData.swift

```diff
@@ -8,46 +8,53 @@
 import Foundation
 
 internal protocol MultipartFormDataBuilder {
-    /// The boundary UUID of this multipart form.
-    var boundary: UUID { get }
+    /// The boundary  of this multipart form.
+    var boundary: String { get }
     /// Adds a field.
     mutating func addFormField(name: String, value: String)
     /// Adds a file.
     mutating func addFormData(name: String, filename: String, data: Data, mimeType: String)
     /// Returns the entire multipart body data (as it should be applied to request).
-    var data: Data { get }
+    mutating func build() -> Data
 }
 
 /// A helper facilitating creation of `multipart/form-data` body.
 internal struct MultipartFormData: MultipartFormDataBuilder {
-    let boundary: UUID
-    private var body = Data()
+    private var body: Data
 
-    init(boundary: UUID) {
-        self.boundary = boundary
+    private(set) var boundary: String
+
+    init(boundary: UUID = UUID()) {
+        self.body = Data()
+        self.boundary = boundary.uuidString
     }
 
     mutating func addFormField(name: String, value: String) {
-        body.append(string: "--\(boundary.uuidString)\r\n")
+        body.append(string: "--\(boundary)\r\n")
         body.append(string: "Content-Disposition: form-data; name=\"\(name)\"\r\n")
         body.append(string: "\r\n")
         body.append(string: value)
         body.append(string: "\r\n")
     }
 
     mutating func addFormData(name: String, filename: String, data: Data, mimeType: String) {
-        body.append(string: "--\(boundary.uuidString)\r\n")
+        body.append(string: "--\(boundary)\r\n")
         body.append(string: "Content-Disposition: form-data; name=\"\(name)\"; filename=\"\(filename)\"\r\n")
         body.append(string: "Content-Type: \(mimeType)\r\n")
         body.append(string: "\r\n")
         body.append(data)
         body.append(string: "\r\n")
     }
 
-    var data: Data {
-        var data = body
-        data.append(string: "--\(boundary.uuidString)--")
-        return data
+    mutating func build() -> Data {
+        defer {
+            // reset builder
+            body = Data()
+            boundary = UUID().uuidString
+        }
```

Comment:
We should reset the builder after build, otherwise we would keep appending data.

Reply:
How was it reset before this change 🤔💭?

Reply:
[We were making a copy of a `struct MultipartFormData` before each request](https://github.com/DataDog/dd-sdk-ios/blob/develop/DatadogSessionReplay/Sources/Feature/RequestBuilders/SegmentRequestBuilder.swift#L39), but we cannot assume that the `protocol MultipartFormDataBuilder` is applied to a value type.